src/client/dfs/dfs.c

```diff
@@ -1054,7 +1054,7 @@ dfs_cont_create(daos_handle_t poh, uuid_t co_uuid, dfs_attr_t *attr,
 	entry.oid.lo = RESERVED_LO;
 	entry.oid.hi = ROOT_HI;
 	daos_obj_generate_id(&entry.oid, 0, dattr.da_oclass_id, 0);
-	entry.mode = S_IFDIR | 0755;
+	entry.mode = S_IFDIR | 0777;
```

Comment:
I'm a little concerned about this. If a user mounts the DFS filesystem on their node, does every other user who accesses the node now have potential read/write access to the files opened with the mounting user's permissions? The user who performed the container open is the one whose permissions are used, since permissions are cached on the handle on the server side.

A possible strategy: The user who opens the container, if allowed access, becomes the "user" for the sake of DFS files/directories. The files' group could still be the container's owner-group. Then default perms could be 775, which offers some protection from rogue users. It's just one idea, I'm open to something else if you have other ideas. Complex ACLs don't translate that well to the standard UNIX user/group/other permissions model, unfortunately.

Reply:
yea i am concerned about that too and wanted to discuss this more.
this really should be 755.
but there is a case where one creates a container, and updates acl to give wr to a user.
if it is 755 user still can't write to that container.
anyway if one has access to the container, all this doesn't matter.
on a shared node, where one mounts the DFS  through dfuse, they won't get to this mountpoint if they don't have access to the container anyway i would assume. maybe @ashleypittman can comment.

Reply:
I think if people are updating ACLs to give access it's not unreasonable to expect them to change the mode as well.

Currently everything in dfuse is confined to a single user (requests by other uid/gids are NACKed by the kernel) but we will be turning off this functionality in future and having / as mode 777 will not be desirable then.

Reply:
OK i will revert this and we can discuss more in the POSIX WG.

but a question:
once you turn off this functionality, on a shared node, when one does:
ls /mnt/dfuse/ or
rm -rf /mnt/dfuse/*
would they be able to do that if they do not have r or rw access access to the container?
meaning would dfuse be using the same global pool/container handle, or would it try to connect to the pool and open the container as the user instead?

Reply:
dfuse will be running as a specific user (it doesn't need root but it won't be the "user") and all I/O will come with the same credentials at the DAOS level, although we can also provide the uid/gid of the user that has made the request.  It'll all be the same pool/container handles.

Reply:
Would it be possible sometime early next week to schedule a call to discuss the usage models for this? I'd like to get a better understanding of what is trying to be done so we can figure out the best way to secure it. I agree with kris though that we definitely lose some security and fidelity in access logging if we start issuing requests to containers under another user's access credentials. I'm not sure what the right answer to that is so I'd like to understand the workflow and usecases better.